render/image.go

```diff
@@ -124,28 +124,39 @@ func drawWatermark(dc *gg.Context) {
 	dc.DrawImageAnchored(watermarkImage, dc.Width()/2, dc.Height()/2, 0.5, 0.5)
```

Comment:
I had to make a lot of updates here. There were some weird interdependencies between how the board structure was indexed and how the coordinates got mapped. The array was iterated incorrectly and the drawing logic compensated for that. This made it difficult to read the code. You had to understand how the board array was filled and iterated to understand why the drawing logic worked.

With the changes here the drawing logic should stand on its own. I also adjusted some naming to make it more clear that we are translating from the board coordinate space to the graphics coordinate space.

The board `(0,0)` is the bottom left, but the graphics `(0,0)` start at the top left.

Reply:
I think this is great! My only note is it might be nice to have the various colors be top-level constants, but that's pretty minor.